File: src/main/java/com/lessmarkup/engine/data/dialects/DatabaseLanguageDialectFactory.java
package com.lessmarkup.engine.data.dialects;

import java.sql.Connection;
import java.sql.SQLException;

public final class DatabaseLanguageDialectFactory {
    public static DatabaseLanguageDialect createDialect(Connection connection) throws SQLException {
        switch (connection.getMetaData().getDriverName()) {
            default:
                return new MySqlDialect();
        }
    }
}

File: src/main/scala/com/lessmarkup/engine/data/dialects/DatabaseLanguageDialectFactory.scala
package com.lessmarkup.engine.data.dialects

import java.sql.Connection

object DatabaseLanguageDialectFactory {
  def createDialect(connection: Option[Connection]): DatabaseLanguageDialect = {

    if (connection.isEmpty) {
      return new MySqlDialect // TODO: create fake dialect class
    }

    val driverName = connection.get.getMetaData.getDriverName

    driverName match {
      case "HSQL Database Engine Driver" => new HsqldbDialect
      case _ => new MySqlDialect
    }
  }
}
